src/refurbished/rocket.f90

```diff
@@ -356,131 +356,207 @@ subroutine update_trajectory( &
+    real(dp) :: acceleration = ZERO
     real(dp) :: altitude = ZERO
-    real(dp) :: area
-    real(dp) :: cv
-    real(dp) :: db = ZERO
+    real(dp) :: burn_depth = ZERO
+    real(dp) :: burn_rate
+    real(dp) :: burning_surface_area
+    real(dp) :: chamber_energy
+    real(dp) :: chamber_mass
+    real(dp) :: chamber_pressure
+    real(dp) :: chamber_temperature
+    real(dp) :: chamber_volume = ONE
     real(dp) :: drag = ZERO
-    real(dp) :: echam
-    real(dp) :: edotgen
-    real(dp) :: edotos
-    real(dp) :: g
+    real(dp) :: energy_generation_rate
+    real(dp) :: energy_outflow_rate
+    real(dp) :: flow_area
+    real(dp) :: fuel_mass = ZERO
+    real(dp) :: heat_capacity_at_constant_volume
+    real(dp) :: heat_capacity_ratio
     integer :: i
-    real(dp) :: mcham
-    real(dp) :: mdotgen
-    real(dp) :: mdotos = ZERO
-    real(dp) :: netthrust = ZERO
-    integer :: nsteps
-    real(dp) :: p
-    real(dp) :: propmass = ZERO
-    real(dp) :: rgas
-    real(dp) :: r
-    real(dp) :: rocketmass = ZERO
-    real(dp) :: surf
-    real(dp) :: t
+    real(dp) :: mass_generation_rate
+    real(dp) :: mass_outflow_rate = ZERO
+    real(dp) :: net_thrust = ZERO
+    integer :: num_time_steps
+    real(dp) :: rocket_mass = ZERO
+    real(dp) :: specific_gas_constant
     real(dp) :: thrust = ZERO
     real(dp) :: time = ZERO
-    real(dp) :: vol = ONE
-    real(dp) :: vel = ZERO
-
-    t = temperature_
-    p = pressure_
-
-    ! propellent grain is a cylinder burning radially outward and axially inward from one end.
-    ! the other end is considered inhibited.
-    ! outer diameter is inhibited because this is a cast propellent: it was poured
-    ! into the tube/chamber and only the inner diameter burns when ignited.
-
-    ! propellant burn rate information
-
-    nsteps = nint(tmax / dt) ! number of time steps
-
-    ! preallocate an output file for simulation infomration
-    allocate(rocket(0:nsteps, 11))
-
-    !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!11
-
-    !! now begin calculating and initializing
-    ! gas variables
-    rgas = UNIVERSAL_GAS_CONSTANT / mw
-    cv = cp - rgas
-    g = cp / cv
-
-    area = PI / 4.0_dp * dia**2 ! nozzle area
-
-    ! calculate initial mass and energy in the chamber
-    mcham = p * vol / rgas / t ! use ideal gas law to determine mass in chamber
-    echam = mcham * cv * t ! initial internal energy in chamber
-
-    rocket(0,:) = [time, p, t, mdotos, thrust, drag, netthrust, vol, accel, vel, altitude]
+    real(dp) :: velocity = ZERO
+
+    chamber_temperature = initial_temperature
+    chamber_pressure = initial_pressure
+    num_time_steps = nint(simulation_end_time / time_step_length)
+    allocate(rocket(0:num_time_steps, 11))
+    specific_gas_constant = UNIVERSAL_GAS_CONSTANT / molecular_weight
+    heat_capacity_at_constant_volume = &
+        heat_capacity_at_constant_pressure - specific_gas_constant
+    heat_capacity_ratio = &
```

Comment:
this is very thorough.  as I mentioned, it's typically referred to as gamma, but your definition is also fine., just a little more lengthy.  especially when you get to calculating mass flow because it's used a  lot.